python/ray/tune/tune.py

```diff
@@ -905,7 +905,7 @@ def run(
     if air_verbosity is None:
         is_trainer = _entrypoint == AirEntrypoint.TRAINER
         progress_reporter = progress_reporter or _detect_reporter(
-            _trainer_api=is_trainer
+            _trainer_api=is_trainer, metric_columns=progress_metrics
```

Comment:
If `progress_metrics` are specified in a user's Trainable class under
`self._progress_metrics`, these never arrive at the default ProgressReporters used by Tuner.

Reply:
It looks like the "new output engine" progress reporter implementation does not use the `progress_metrics` argument at all:

https://github.com/ray-project/ray/blob/ab8e10a4475f9af4c6de80d9fb6e3915da92bb3a/python/ray/tune/experimental/output.py#L606

Reply:
The problem is that the "new output engine" reporter [only expects a set of metric columns to show](https://github.com/ray-project/ray/blob/ab8e10a4475f9af4c6de80d9fb6e3915da92bb3a/python/ray/tune/experimental/output.py#L684), and [doesn't support this renaming feature anymore](https://github.com/ray-project/ray/blob/bbcdc496d69ce1c35154f1eeb0aa5d7ea916196c/python/ray/tune/experimental/output.py#L558). It only uses the keys of `_progress_metrics` to determine which columns to print out in the display table.

The old output engine goes through this `CLIReporter` codepath which does implement the renaming dict: https://github.com/ray-project/ray/blob/fea11844019c00e0d098d5b0ad92814b58957e96/python/ray/tune/progress_reporter.py#L817-L820

Reply:
The solution here is to re-implement it in the new codepath. We can add a renaming dict kwarg in this function: https://github.com/ray-project/ray/blob/bbcdc496d69ce1c35154f1eeb0aa5d7ea916196c/python/ray/tune/experimental/output.py#L555

Then, the usage [here](https://github.com/ray-project/ray/blob/bbcdc496d69ce1c35154f1eeb0aa5d7ea916196c/python/ray/tune/experimental/output.py#L681) will be updated:

```python
            _print_dict_as_table(
                result,
                header=f"{self._addressing_tmpl.format(trial)} result",
                include=list(self._progress_metrics),
                exclude=BLACKLISTED_KEYS,
                division=AUTO_RESULT_KEYS,
                rename=self._progress_metrics,
            )
```

Let me know if you need help including this in the PR.

Reply:
Hey @justinvyu , ok, but it's not just about the renaming feature not being available in the "new output engine". It's that even without any renaming (`_progress_metrics` is a tuple), the provided metrics to print do NOT reach the default reporter. Instead, Tune overrides these with its own DEFAULT_METRICS (which don't fit well for RL experiments).

I'll take another look at the code you sent above (thanks for pointing me to the right direction!) and let you know, if I get stuck.

Reply:
Could you point me to where the override happens? When I tested with a custom `_progress_metrics`, Tune was able to propagate it correctly.